visualization/app/codeCharta/util/fileValidator.ts

```diff
@@ -22,23 +75,43 @@ export class FileValidator {
 		return true
 	}
 
-	public static validate(file: { nodes: CodeMapNode[] }): Array<{ message: string; dataPath: string }> {
-		if (!file) {
-			return [{ message: "file is empty or invalid", dataPath: "empty or invalid file" }]
+	private static getFileApiVersion(file: { nodes: CodeMapNode[] }) {
+		for (let key in file) {
+			if (key === "apiVersion") {
+				return file[key]
+			}
 		}
-		let ajv = require("ajv")()
-		let compare = ajv.compile(require("./schema.json"))
-		let valid = compare(file)
+	}
 
-		if (!FileValidator.hasUniqueChildren(file.nodes[0])) {
-			return [
-				{
-					message: "names or ids are not unique",
-					dataPath: "uniqueness"
-				}
-			]
-		}
+	private static isValidApiVersion(file: { nodes: CodeMapNode[] }): boolean {
+		const apiVersion = this.getFileApiVersion(file)
+		const hasApiVersion = apiVersion !== undefined
+		const versionRegExp = new RegExp("[0-9]+.[0-9]+")
```

Comment:
A dot `.` can be any character, while `\.` is only a dot.
```suggestion
		const versionRegExp = new RegExp("[0-9]+\.[0-9]+")
```

Reply:
It seems the \ gets removed by the prehook

Reply:
We need to use \\ to escape the escape character as the whole thing will be converted to string later, should be fixed now